integration-healthcheck/src/main/java/org/symphonyoss/integration/healthcheck/services/ServiceHealthIndicator.java

```diff
@@ -0,0 +1,213 @@
+import javax.ws.rs.ProcessingException;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import static javax.ws.rs.core.Response.Status.OK;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Abstract class that holds common methods to all service health indicators.
+ *
+ * Created by rsanchez on 27/01/17.
+ */
+public abstract class ServiceHealthIndicator implements HealthIndicator {
+
+  private static final ISymphonyLogger LOG =
+      IntegrationBridgeCloudLoggerFactory.getLogger(ServiceHealthIndicator.class);
+
+  /**
+   * Version field
+   */
+  private static final String VERSION = "version";
+
+  /**
+   * HTTP Connection timeout (in miliseconds)
+   */
+  private static final int CONNECT_TIMEOUT_MILLIS = 1000;
+
+  /**
+   * HTTP Read timeout (in miliseconds)
+   */
+  private static final int READ_TIMEOUT_MILLIS = 5000;
+
+  /**
+   * Cache period (in seconds) to retrive the service information.
+   */
+  private static final int SERVICE_CACHE_PERIOD_SECS = 20;
+
+  @Autowired
+  protected IntegrationProperties properties;
+
+  @Autowired
+  private AuthenticationProxy authenticationProxy;
+
+  /**
+   * Cache for the service information.
+   */
+  private LoadingCache<String, IntegrationBridgeService> serviceInfoCache;
+
+  @PostConstruct
+  public void init() {
+    serviceInfoCache = CacheBuilder.newBuilder().expireAfterWrite(SERVICE_CACHE_PERIOD_SECS,
+        TimeUnit.SECONDS).build(new CacheLoader<String, IntegrationBridgeService>() {
+      @Override
+      public IntegrationBridgeService load(String key) throws Exception {
+        if (key.equals(getServiceName())) {
+          return getServiceInfo();
+        }
+
+        return null;
+      }
+    });
+  }
+
+  @Override
+  public Health health() {
+    String serviceName = getServiceName();
+
+    try {
+      IntegrationBridgeService service = serviceInfoCache.get(serviceName);
+      return Health.status(service.getConnectivity()).withDetail(serviceName, service).build();
+    } catch (UncheckedExecutionException | ExecutionException e) {
+      LOG.error(String.format("Unable to retrieve %s info", serviceName), e);
+      return Health.unknown().build();
+    }
+  }
+
+  /**
+   * Retrieves the service information like connectivity, current version, and compatibility.
+   * @return Service information
+   */
+  private IntegrationBridgeService getServiceInfo() {
+    IntegrationBridgeService service = new IntegrationBridgeService(getMinVersion());
+
+    String healthResponse = getHealthResponse();
+
+    if (healthResponse == null) {
+      service.setConnectivity(Status.DOWN);
+    } else {
+      service.setConnectivity(Status.UP);
+
+      try {
+        JsonNode node = JsonUtils.readTree(healthResponse);
+        String version = node.path(VERSION).asText();
```

Comment:
Does this work for all services?  Shouldn't each implementation of ServiceHealthIndicator be responsible for defining where in the response the version should be found?